autogpt_platform/frontend/src/app/(platform)/library/agents/[id]/components/AgentRunsView/components/RunAgentInputs/RunAgentInputs.tsx

```diff
@@ -43,7 +46,14 @@ export function RunAgentInputs({
 }: Props & React.HTMLAttributes<HTMLElement>) {
   const { handleUploadFile, uploadProgress } = useRunAgentInputs();
 
-  const dataType = determineDataType(schema);
+  let dataType = determineDataType(schema);
+
+  // Override dataType if format is "table"
+  // This is needed because determineDataType doesn't always see the format field
+  if ("format" in schema && (schema as any).format === "table") {
+    dataType = DataType.TABLE;
+  }
```

Comment:
why not amend `determineDataType` instead?

Reply:
lol good point